app/views/assignments/_assignment_form_options.html.erb

```diff
@@ -16,7 +16,7 @@
 <dl class="<%= view.form_class_for(:public_repo) %> my-4">
   <div class="form-checkbox">
     <label>
-      <%= f.radio_button :public_repo, true %>
+      <%= f.radio_button :visibility, "public", checked: !view.private_repos_available? %>
```

Comment:
This partial also shows up when you click "Assignment settings". Will the `checked` override the actual current visibility of an assignment?

Reply:
I'll change this to look at the assignment's visibility if the assignment has been persisted!